addons/website_livechat/models/mail_channel.py

```diff
@@ -36,7 +36,7 @@ def channel_info(self, extra_info=False):
                     'name': visitor.display_name,
                     'country_code': visitor.country_id.code.lower() if visitor.country_id else False,
                     'is_connected': visitor.is_connected,
-                    'history': self._get_visitor_history(visitor),
+                    'history': self.sudo()._get_visitor_history(visitor),
```

Comment:
double check, this should not be introduced here if we are restoring a feature that supposedly existed and worked

Reply:
I fail to see which change from 13.3 to 13.4 causes the access right issue.
To reproduce: invite demo user in a livechat.
Access right error here:
https://github.com/odoo/odoo/blob/saas-13.4/addons/website_livechat/models/mail_channel.py#L53
_(`page_id.name` raises the error)_

Same code in 13.3 without access error raised (and same data). Demo user seems to be in same groups, and can't find ACL change on `website.page`.

Reply:
probably @dbeguin or @awa-odoo could help you here, if they modified this code recently?

Reply:
not recently, but we needed to give read access to livechat users to website.page and ui.ir.view in order to see the website visitor tree list from livechat menu. (But I couldn't find the cause either..)
see https://github.com/odoo/odoo/pull/56678 --> https://github.com/odoo/odoo/pull/56678/commits/bdea5e6fdbdb7f526d5023a2139875b2ea744219(not merged yet)

Reply:
This is the commit in 13.4 that makes this `sudo` necessary:

https://github.com/odoo/odoo/commit/ccc98e01691b88c2da9c484c0e6de235b9d18548#diff-2516dc5a739891abf193cd9902befc65

Here I think it's better to add `sudo` to this specific flow than change ACL of all livechat users to see all pages and views though.
If you think it is necessary in the context of your other fix, feel free to change it though.